src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
+            if (intersection.size() == 2)
+            {
+                node = node_based_graph.GetTarget(sliproad_edge);
+                sliproad_edge = intersection[1].eid;
+                intersection = intersection_generator(node, sliproad_edge);
+            }
+            return intersection;
+        }(intersection_node_id);
+
+        const NodeID sliproad_edge_target = node_based_graph.GetTarget(sliproad_edge);
+
+        // Distinct triangle nodes `bcd`
+        if (intersection_node_id == next->node || intersection_node_id == sliproad_edge_target ||
+            next->node == sliproad_edge_target)
         {
-            EdgeID candidate_in = road.eid;
-            const auto target_intersection = [&](NodeID node) {
-                auto intersection = intersection_generator(node, candidate_in);
-                // skip over traffic lights
-                if (intersection.size() == 2)
-                {
-                    node = node_based_graph.GetTarget(candidate_in);
-                    candidate_in = intersection[1].eid;
-                    intersection = intersection_generator(node, candidate_in);
-                }
-                return intersection;
-            }(intersection_node_id);
-
-            const auto link_data = node_based_graph.GetEdgeData(road.eid);
-            // Check if the road continues here
-            const bool is_through_street =
-                !target_intersection.empty() && link_data.name_id != EMPTY_NAMEID &&
-                target_intersection.end() !=
-                    std::find_if(
-                        target_intersection.begin() + 1,
-                        target_intersection.end(),
-                        [this, &link_data](const ConnectedRoad &road) {
-                            const auto &road_edge_data = node_based_graph.GetEdgeData(road.eid);
-
-                            const auto same_name =
-                                road_edge_data.name_id != EMPTY_NAMEID &&
-                                !util::guidance::requiresNameAnnounced(road_edge_data.name_id,
-                                                                       link_data.name_id,
-                                                                       name_table,
-                                                                       street_name_suffix_table);
-
-                            return same_name;
-                        });
-
-            // if the sliproad candidate is a through street, we cannot handle it as a sliproad
-            if (is_through_street)
+            continue;
+        }
+
+        // If the sliproad candidate is a through street, we cannot handle it as a sliproad.
+        if (isThroughStreet(sliproad_edge, target_intersection))
+        {
+            continue;
+        }
+
+        // The turn off of the Sliproad has to be obvious and a narrow turn
+        {
+            const auto index = findObviousTurn(sliproad_edge, target_intersection);
```

Comment:
I'd go as far as to say it has to be the `only` turn available. Did you find any coincidences where larger intersections would be desirable?